### ISSUE TITLE
Tncart does not work with Ghoauld

### ISSUE THREAD

**Issue by xKNIGHTMAREx** [buppu3/tnCart#24]
I have a VG8020 that I have been able to revive thanks to Ghoauld, it works perfectly with all the cartridges I have tried and with the Wondertang, now with the Tncartwonder when it starts up the first screen that detects the SD appears, the image of the nextor.sys and it goes to Basic. Is there any way to fix this? I would like to be able to use the tncart together with the Ghoauld. The PCB is 2.0b. Thanks.

**Comment by buppu3** [comment]
You wrote in issue#14, is this a separate issue from that?
Is it possible to get a list of files with the FILES command after starting BASIC?

**Comment by xKNIGHTMAREx** [comment]
Then I try to do it and see what happens.

**Comment by xKNIGHTMAREx** [comment]
It says Files not found

**Comment by xKNIGHTMAREx** [comment]
https://github.com/user-attachments/assets/dd868715-1963-47da-a41b-c39fb9494a40

**Comment by xKNIGHTMAREx** [comment]
The interference is due to the Ghoauld HDMI to VGA adapter

**Comment by xKNIGHTMAREx** [comment]
This video is loading the wondertang


https://github.com/user-attachments/assets/37e12153-1f60-43b9-874f-2e624a636250

**Comment by Chandler-Kluser** [comment]
[@jabadia](https://github.com/jabadiagm)

**Comment by buppu3** [comment]
> It says Files not found

Is the response to the FILES command “file not found” instead of “i/o error” etc.?

**Comment by polileco** [comment]
Are you using the same uSD for both cores on the same cartridge? And you did the partition on uSD using WonderTANG? 

Can you tell what is the Nextor kernel version of both TnCart and WonderTANG?

What device and partitions "CALL FDISK" command show when runing from BASIC from both cores?

Regards

**Comment by xKNIGHTMAREx** [comment]
The version I put in the Tncartwonder is the one on GitHub and for the wondertang the one on its GitHub

**Comment by xKNIGHTMAREx** [comment]
Nextor 2.1.2 for Tncartwonder and Nextor 2.1.1 for Wondertang

**Comment by xKNIGHTMAREx** [comment]
When I put files now this appears
![Uploading IMG20241109154639.jpg…]()

**Comment by polileco** [comment]
How about FDISK?

If the uSD is the same for both cores maybe you can try downgrade only nextor kernel to 2.1.1 on TnCart.

**Comment by xKNIGHTMAREx** [comment]
If I run Call Fdisk with the Wondertang it lets me format the SD and create a 4G partition, but if I do it with the one that Tncartwonder has, it only lets me create a 128mb partition.

**Comment by xKNIGHTMAREx** [comment]
Uploading VID-20241109-WA0036.mp4…

**Comment by xKNIGHTMAREx** [comment]
https://github.com/user-attachments/assets/51789c32-5888-4f8f-ad15-3d5c0c72283c

**Comment by xKNIGHTMAREx** [comment]
However, from the wondertang it does make the files for me
![IMG-20241109-WA0037](https://github.com/user-attachments/assets/a3b8f307-9e34-4d70-afc7-0951c72ed93e)

**Comment by xKNIGHTMAREx** [comment]
I have tried with nextor 2.1.0 and 2.1.1 megaflashsdscc-1.slot.rom and it does the same.

**Comment by buppu3** [comment]
Both tnCart and goauld multiplex the bus and the timings may not match.
Asserting the WAIT signal on the Z80 during TF access is expected to reduce the amount of data missed.
The WAIT signal can be controlled by synthesizing the USE_WAIT_SIGNAL parameter of the TF_CONTROLLER module with 1.

**Comment by xKNIGHTMAREx** [comment]
How do I put 1 in that module?

**Comment by buppu3** [comment]
If you want it right away, you have to install Gowin FPGA Designer, modify a part of tnCartWonder, and synthesize the bitstream file.
The part to change is USE_WAIT_SIGNAL = 0 at line 40 of "rtl/src/peripheral/msx/tf_controller.sv" in the tnCartWonder repository.

**Comment by xKNIGHTMAREx** [comment]
I'll try to do it, although I don't know if I'll be able to do it well.

**Comment by Chandler-Kluser** [comment]
> Both tnCart and goauld multiplex the bus and the timings may not match. Asserting the WAIT signal on the Z80 during TF access is expected to reduce the amount of data missed. The WAIT signal can be controlled by synthesizing the USE_WAIT_SIGNAL parameter of the TF_CONTROLLER module with 1.

well, goa'uld works with wondertang, which actually also multiplexes address bus...

**Comment by xKNIGHTMAREx** [comment]
The Wondertang 2.0 works perfectly with the Ghoauld, but the Tncartwonder on the 2.0 board doesn't, it boots up the computer and always goes to Basic. Could someone modify the .fs and put the file on GitHub? I've tried to do it but I have no idea, thanks a lot.

**Comment by buppu3** [comment]
> > Both tnCart and goauld multiplex the bus and the timings may not match. Asserting the WAIT signal on the Z80 during TF access is expected to reduce the amount of data missed. The WAIT signal can be controlled by synthesizing the USE_WAIT_SIGNAL parameter of the TF_CONTROLLER module with 1.
> 
> well, goa'uld works with wondertang, which actually also multiplexes address bus...

WonderTANG and tnCart have completely different Nextor driver implementation methods, which is probably why this phenomenon occurs.
WonderTANG is a highly functional implementation that performs initial card settings, etc. within the FPGA, while tnCart only performs SPI conversion and initial settings, etc. are performed by the MSX driver software.
The tnCart may have failed to read the CSD during the initialization process, causing it to behave strangely.

**Comment by xKNIGHTMAREx** [comment]
Y modificando está línea "USE_WAIT_SIGNAL = 0" a 1 funcionará?. Podrías modificarlo y así lo pruebo, te lo agradecería mucho. Gracias 😊.

**Comment by xKNIGHTMAREx** [comment]
And modifying this line "USE_WAIT_SIGNAL = 0" to 1 will work? I will try to modify it to see if it works.

**Comment by buppu3** [comment]
> And modifying this line "USE_WAIT_SIGNAL = 0" to 1 will work? I will try to modify it to see if it works.

I do not know why Nextor does not work on your machine, so I cannot say for sure if it will work with this fix.
I suggested this as one possible cause.

**Comment by xKNIGHTMAREx** [comment]
I'll try it and comment here.

**Comment by xKNIGHTMAREx** [comment]
I changed the parameter 0 to 1 and it still doesn't work. I guess I did the whole process right. I changed the 0 to 1 and after synchronizing I put the .fs in the Tncartwonder.

https://github.com/user-attachments/assets/e3536435-86ad-436a-bc51-45b5c3f9a006

**Comment by buppu3** [comment]
> I changed the parameter 0 to 1 and it still doesn't work. I guess I did the whole process right. I changed the 0 to 1 and after synchronizing I put the .fs in the Tncartwonder.
> 
> https://github.com/user-attachments/assets/e3536435-86ad-436a-bc51-45b5c3f9a006
> 
> 

Are you flashing the files created by the rebuild?
New files will be created in "rtl/impl/pnr/"

Also, do functions other than Nextor work?

**Comment by xKNIGHTMAREx** [comment]
Yes, use the .fs from that folder, it is supposed to be the one you created. I have problems with the computer where I have Ghoauld, until I fix it I will not be able to continue testing. Once it is fixed we will continue testing.
